### PR TITLE
EKF2 move most orb subscriptions to uORB::Subscription

### PR DESCRIPTION
**Pull request by dagar**
Requires https://github.com/PX4/Firmware/pull/12040 before merging.

### CODE REVIEW

**Comment by dagar** [comment]
Ready for review and testing.

File: src/modules/ekf2/ekf2_main.cpp [hunk 1]
```diff
@@ -1212,18 +1147,27 @@ void Ekf2::run()
...
-			_range_finder_sub_index = getRangeSubIndex(_range_finder_subs);
+			for (unsigned i = 0; i < ORB_MULTI_MAX_INSTANCES; i++) {
+				if (_range_finder_subs[i].updated()) {
+					distance_sensor_s report{};
+					_range_finder_subs[i].copy(&report);
+
+					// only use the first instace which has the correct orientation
+					if (report.orientation == distance_sensor_s::ROTATION_DOWNWARD_FACING) {
+						PX4_INFO("Found range finder with instance %d", i);
+						_range_finder_sub_index = i;
```
Comment (julianoes) [inline src/modules/ekf2/ekf2_main.cpp]: I think having this encapsulated in a function was nicer, just give it a better name so it says what it does.

**Comment by jorge789** [comment]
### Tested on Pixhawk 4 V5
**Modes Tested**

Position Mode: Good.
Altitude Mode: Good.
Stabilized Mode: Good.
Mission Plan Mode (Automated): Good.
RTL: Good.

**- Procedure**
Arm and Take off in position mode, after flying for approximately one minute, switched to altitude then stabilized mode proceed to switch to mission plan mode then make sure that vehicle follows all waypoints as shown in QGC, once completed all waypoin activate RTL and see landing behaviour.

**Notes:**
No issues noted, good flight in general.

**Logs:**
https://review.px4.io/plot_app?log=26917c92-5aa0-4c10-bb11-a1ebebb5770f

https://review.px4.io/plot_app?log=cec863bf-1fcb-4dd5-b0d1-27673d50de96

**COMPARISON OF MASTER :**
https://review.px4.io/plot_app?log=1411598d-8567-48fc-844e-6a54d9b302a4

File: src/modules/ekf2/ekf2_main.cpp (hunk 1 above)
Comment (dagar) [reply src/modules/ekf2/ekf2_main.cpp]: Makes sense, restored.

**Pull request merged**